pkg/netpol/connlist/connlist.go

```diff
@@ -391,27 +417,35 @@ func (ca *ConnlistAnalyzer) getConnectionsList(pe *eval.PolicyEngine, ia *ingres
 	}
 	connsRes = append(connsRes, ingressAllowedConns...)
 
-	if len(peersAllowedConns) == 0 {
+	if ca.focusWorkload == "" && len(peersAllowedConns) == 0 {
 		ca.logger.Warnf("connectivity analysis found no allowed connectivity between pairs from the configured workloads or external IP-blocks")
 	}
 
-	return connsRes, peersRes, nil
+	return connsRes, peers, nil
 }
 
-// getConnectionsBetweenPeers returns connections list from PolicyEngine object
-func (ca *ConnlistAnalyzer) getConnectionsBetweenPeers(pe *eval.PolicyEngine) ([]Peer2PeerConnection, []Peer, error) {
-	// get workload peers and ip blocks
-	peerList, err := pe.GetPeersList()
-	if err != nil {
-		ca.errors = append(ca.errors, newResourceEvaluationError(err))
-		return nil, nil, err
+// existsFocusWorkload checks if the provided focus workload is ingress-controller
+// or if it exists in the peers list from the parsed resources
+func (ca *ConnlistAnalyzer) existsFocusWorkload(peerList []eval.Peer, excludeIA bool) bool {
+	// if focus workload is ingress controller, it is okay to continue checking for connections
+	ingressPodNsNameFormat := types.NamespacedName{Namespace: ingressanalyzer.IngressPodNamespace,
+		Name: ingressanalyzer.IngressPodName}.String()
+	if ca.focusWorkload == ingressanalyzer.IngressPodName || ca.focusWorkload == ingressPodNsNameFormat {
+		return !excludeIA // if the ingress-analyzer is empty,
```

Comment:
maybe the warning should be adjusted in this case... it's not "a missing workload from the input", it's : the "ingress controller was not added to the analysis, since Ingress/Route resources were not found"

Reply:
handled